src/webgpu/api/validation/query_set/create.spec.ts

```diff
@@ -2,140 +2,110 @@ export const description = `
 Tests for validation in createQuerySet.
 `;
 
-import { poptions } from '../../../../common/framework/params_builder.js';
+import { params, poptions } from '../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
 import { ValidationTest } from '../validation_test.js';
 
 export const g = makeTestGroup(ValidationTest);
 
 const kMaxQueryCount = 8192;
+const kQuerytype = ['occlusion', 'pipeline-statistics', 'timestamp'] as const;
 
-// Test with these query counts.
-const kQueryCounts: number[] = [
-  // valid
-  0,
-  kMaxQueryCount,
-  // invalid
-  kMaxQueryCount + 1,
-];
-
-g.test('occlusion')
+g.test('count')
   .desc(
     `
-Tests that create query set with the type of occlusion.
-- control cases:
-  - count <= kMaxQueryCount
-  - pipelineStatistics is undefined or empty
-- invalid cases:
-  - count > kMaxQueryCount
-  - pipelineStatistics is set with pipeline statistic names
+Tests that create query set with the count for all query types:
+- count {<, =, >} kMaxQueryCount
+- x= {occlusion, pipeline-statistics, timestamp} query
   `
   )
-  .params([
-    ...poptions('count', kQueryCounts),
-    ...poptions('pipelineStatistics', [
-      // valid
-      undefined,
-      [],
-      // invalid
-      ['clipper-invocations'],
-    ]),
-  ])
-  .fn(t => {
-    const { count = 1, pipelineStatistics } = t.params as {
-      count?: number;
-      pipelineStatistics?: GPUPipelineStatisticName[];
-    };
+  .params(
+    params()
+      .combine(poptions('type', kQuerytype))
+      .combine(poptions('count', [0, kMaxQueryCount, kMaxQueryCount + 1]))
+  )
+  .fn(async t => {
+    const { type, count } = t.params;
+
+    if (type === 'pipeline-statistics') {
+      await t.selectDeviceOrSkipTestCase({
+        extensions: ['pipeline-statistics-query'],
+      });
+    } else if (type === 'timestamp') {
+      await t.selectDeviceOrSkipTestCase({
+        extensions: ['timestamp-query'],
+      });
+    }
```

Comment:
Perhaps consider making a helper to deduplicate this code (`async function selectDeviceForQueryType(t: GPUTest, type: GPUQueryType): Promise<void>` in this file)

Reply:
Added the helper function.